pubsub/api/src/update_topic_type.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Copyright 2025 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * For instructions on how to run the full sample:
+ *
+ * @see https://github.com/GoogleCloudPlatform/php-docs-samples/blob/main/pubsub/api/README.md
+ */
+
+namespace Google\Cloud\Samples\PubSub;
+
+# [START pubsub_update_topic_type]
+use Google\Cloud\PubSub\PubSubClient;
+
+/**
+ * Changes the type of a Pub/Sub topic.
+ *
+ * @param string $projectId  The Google project ID.
+ * @param string $topicName  The Pub/Sub topic name.
+ * @param string $streamArn  The Kinesis stream ARN to ingest data from.
+ * @param string $consumerArn  The Kinesis consumer ARN to used for ingestion in Enhanced Fan-Out mode. The consumer must be already created and ready to be used.
+ * @param string $awsRoleArn  AWS role ARN to be used for Federated Identity authentication with Kinesis. Check the Pub/Sub docs for how to set up this role and the required permissions that need to be attached to it.
+ * @param string $gcpServiceAccount  The GCP service account to be used for Federated Identity authentication with Kinesis (via a AssumeRoleWithWebIdentity call for the provided role). The $awsRoleArn must be set up with accounts.google.com:sub equals to this service account number.
+ */
+function update_topic_type(
+    string $projectId,
+    string $topicName,
+    string $streamArn,
+    string $consumerArn,
+    string $awsRoleArn,
+    string $gcpServiceAccount
+): void {
+    $pubsub = new PubSubClient([
+        'projectId' => $projectId,
+    ]);
+
+    $topic = $pubsub->topic($topicName, [
+        'ingestionDataSourceSettings' => [
+            'aws_kinesis' => [
+                'stream_arn' => $streamArn,
+                'consumer_arn' => $consumerArn,
+                'aws_role_arn' => $awsRoleArn,
+                'gcp_service_account' => $gcpServiceAccount
+            ]
+        ]
+    ]);
+
+    $topic->update([], [
+        'updateMask' => [ 'ingestionDataSourceSettings' ]
+    ]);
```

Comment:
This is a valid way to do it, but I think when _updating_, the API would normally look a bit more like this:

```suggestion
    $topic = $pubsub->topic($topicName);
    $topic->update([
        'ingestionDataSourceSettings' => [
            'aws_kinesis' => [
                'stream_arn' => $streamArn,
                'consumer_arn' => $consumerArn,
                'aws_role_arn' => $awsRoleArn,
                'gcp_service_account' => $gcpServiceAccount
            ]
        ]
    ], [
        'updateMask' => [
            'ingestionDataSourceSettings'
        ]
    ]);

```

Then the `update` function will calculate the update mask for you.